google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaData.java

```diff
@@ -0,0 +1,1400 @@
+            StructField.of("SUPERTABLE_NAME", Type.string())
+        ),
+        Collections.<Struct>emptyList()));
+  }
+
+
+  @Override
+  public ResultSet getAttributes(String catalog, String schemaPattern, String typeNamePattern,
+      String attributeNamePattern) throws SQLException {
+    return JdbcResultSet.of(ResultSets.forRows(
+        Type.struct(
+            StructField.of("TYPE_CAT", Type.string()),
+            StructField.of("TYPE_SCHEM", Type.string()),
+            StructField.of("TYPE_NAME", Type.string()),
+            StructField.of("ATTR_NAME", Type.string()),
+            StructField.of("DATA_TYPE", Type.int64()),
+            StructField.of("ATTR_TYPE_NAME", Type.string()),
+            StructField.of("ATTR_SIZE", Type.int64()),
+            StructField.of("DECIMAL_DIGITS", Type.int64()),
+            StructField.of("NUM_PREC_RADIX", Type.int64()),
+            StructField.of("NULLABLE", Type.int64()),
+            StructField.of("REMARKS", Type.string()),
+            StructField.of("ATTR_DEF", Type.string()),
+            StructField.of("SQL_DATA_TYPE", Type.int64()),
+            StructField.of("SQL_DATETIME_SUB", Type.int64()),
+            StructField.of("CHAR_OCTET_LENGTH", Type.int64()),
+            StructField.of("ORDINAL_POSITION", Type.int64()),
+            StructField.of("IS_NULLABLE", Type.string()),
+            StructField.of("SCOPE_CATALOG", Type.string()),
+            StructField.of("SCOPE_SCHEMA", Type.string()),
+            StructField.of("SCOPE_TABLE", Type.string()),
+            StructField.of("SOURCE_DATA_TYPE", Type.int64())
+        ),
+        Collections.<Struct>emptyList()));
+  }
+
+  @Override
+  public boolean supportsResultSetHoldability(int holdability) throws SQLException {
+    return holdability == ResultSet.CLOSE_CURSORS_AT_COMMIT;
+  }
+
+  @Override
+  public int getResultSetHoldability() throws SQLException {
+    return ResultSet.CLOSE_CURSORS_AT_COMMIT;
+  }
+
+  @Override
+  public int getDatabaseMajorVersion() throws SQLException {
+    return DATABASE_MAJOR_VERSION;
+  }
+
+  @Override
+  public int getDatabaseMinorVersion() throws SQLException {
+    return DATABASE_MINOR_VERSION;
+  }
+
+  @Override
+  public int getJDBCMajorVersion() throws SQLException {
+    return JDBC_MAJOR_VERSION;
+  }
+
+  @Override
+  public int getJDBCMinorVersion() throws SQLException {
+    return JDBC_MINOR_VERSION;
+  }
+
+  @Override
+  public int getSQLStateType() throws SQLException {
+    return sqlStateSQL;
+  }
+
+  @Override
+  public boolean locatorsUpdateCopy() throws SQLException {
+    return true;
+  }
+
+  @Override
+  public boolean supportsStatementPooling() throws SQLException {
+    return false;
+  }
+
+  @Override
+  public RowIdLifetime getRowIdLifetime() throws SQLException {
+    return RowIdLifetime.ROWID_UNSUPPORTED;
+  }
+
+  @Override
+  public ResultSet getSchemas(String catalog, String schemaPattern) throws SQLException {
```

Comment:
do we have tests for this (and the other sql query methods)?

Reply:
Yes, but they are created as integration tests (`ITJdbcDatabaseMetaDataTest` in the private repo), so they will be added in a later PR.